src/mongocxx/test/spec/change_stream.cpp

```diff
@@ -122,47 +133,50 @@ void run_change_stream_tests_in_file(const std::string& test_path) {
                 op_runner.run(operation.get_document().value);
             }
-        }
 
-        // "Wait until either: (1) An error occurs (2) All operations have been successful"
-        bool had_error = false;
-        auto expected_result = test["result"].get_document().value;
-        std::vector<document::value> changes;
-        try {
-            for (auto&& change : cs) {
-                /* store a copy of the event. */
-                changes.emplace_back(document::value(change));
-            }
-        } catch (operation_exception& oe) {
-            REQUIRE(expected_result["error"]);
-            auto actual_error = oe.raw_server_error();
-            REQUIRE(actual_error);
-            // "Assert that the error MATCHES results.error"
-            REQUIRE(matches(actual_error->view(), expected_result["error"].get_document().value));
-            had_error = true;
-        };
-
-        // "Assert that the changes received from changeStream MATCH the results in results.success"
-        if (!had_error) {
-            REQUIRE(expected_result["success"]);
-            for (auto&& expected_change : expected_result["success"].get_array().value) {
-                auto expected_change_view = expected_change.get_document().value;
-                bool found = false;
-
-                for (auto&& change : changes) {
-                    if (matches(change.view(), expected_change_view)) {
-                        found = true;
-                        break;
-                    }
+            // "Wait until either: (1) An error occurs (2) All operations have been successful"
+            bool had_error = false;
+            auto expected_result = test["result"].get_document().value;
+            std::vector<document::value> changes;
+            try {
+                for (auto&& change : cs) {
+                    /* store a copy of the event. */
+                    changes.emplace_back(document::value(change));
+                }
+            } catch (operation_exception& oe) {
+                REQUIRE(expected_result["error"]);
+                auto actual_error = oe.raw_server_error();
+                REQUIRE(actual_error);
+                // "Assert that the error MATCHES results.error"
+                REQUIRE(
+                    matches(actual_error->view(), expected_result["error"].get_document().value));
+                had_error = true;
+            };
+
+            // "Assert that the changes received from changeStream MATCH the results in
+            // results.success"
+            if (!had_error) {
+                REQUIRE(expected_result["success"]);
+                for (auto&& expected_change : expected_result["success"].get_array().value) {
+                    REQUIRE(std::count_if(
+                        changes.begin(), changes.end(), [&](const document::value& res) {
+                            return matches(res.view(), expected_change.get_document().value);
+                        }));
```

Comment:
Same logic as before but did a bit of refactoring.

Reply:
I guess using `find_if` would save us from having to iterate through the entire list. Done